packages/headless-components/forms/src/react/Form.tsx

```diff
@@ -0,0 +1,745 @@
+import {
+  CheckboxGroupProps,
+  CheckboxProps,
+  PhoneInputProps,
+  DateInputProps,
+  DatePickerProps,
+  DateTimeInputProps,
+  DropdownProps,
+  FileUploadProps,
+  MultilineAddressProps,
+  NumberInputProps,
+  RadioGroupProps,
+  RatingInputProps,
+  RichTextProps,
+  SignatureProps,
+  SubmitButtonProps,
+  TagsProps,
+  TextAreaProps,
+  TextInputProps,
+  TimeInputProps,
+  ProductListProps,
+  FixedPaymentProps,
+  PaymentInputProps,
+  DonationProps,
+  AppointmentProps,
+  ImageChoiceProps,
+} from './types.js';
+
+enum TestIds {
+  formLoadingError = 'form-loading-error',
+  formError = 'form-error',
+  formSubmitted = 'form-submitted',
+}
+
+/**
+ * Props for the Form root component following the documented API
+ */
+export interface RootProps {
+  children: React.ReactNode;
+  /** Form service configuration */
+  formServiceConfig: FormServiceConfig;
+  className?: string;
+}
+
+/**
+ * Root component that provides all necessary service contexts for a complete form experience.
+ * This component sets up the Form service and provides context to child components.
+ * Must be used as the top-level component for all form functionality.
+ *
+ * @order 1
+ * @component
+ * @param {RootProps} props - Component props
+ * @param {React.ReactNode} props.children - Child components that will have access to form context
+ * @param {forms.Form} props.form - Form object
+ * @param {string} [props.className] - CSS classes to apply to the root element
+ * @example
+ * ```tsx
+ * import { Form } from '@wix/headless-forms/react';
+ *
+ * const FIELD_MAP = {
+ *   TEXT_INPUT: TextInput,
+ *   TEXT_AREA: TextArea,
+ *   CHECKBOX: Checkbox,
+ *   // ... other field components
+ * };
+ *
+ * function FormPage({ formServiceConfig }) {
+ *   return (
+ *     <Form.Root formServiceConfig={formServiceConfig}>
+ *       <Form.Loading>
+ *         {() => (
+ *           <div className="flex justify-center p-4">
+ *             <div>Loading form...</div>
+ *           </div>
+ *         )}
+ *       </Form.Loading>
+ *       <Form.LoadingError>
+ *         {({ error }) => (
+ *           <div className="bg-background border-foreground text-foreground px-4 py-3 rounded mb-4">
+ *             {error}
+ *           </div>
+ *         )}
+ *       </Form.LoadingError>
+ *       <Form.Fields fieldMap={FIELD_MAP} />
+ *     </Form.Root>
+ *   );
+ * }
+ * ```
+ */
+export const Root = React.forwardRef<HTMLDivElement, RootProps>(
+  (props, ref) => {
+    const { children, formServiceConfig, className } = props;
+
+    return (
+      <CoreForm.Root formServiceConfig={formServiceConfig}>
+        <RootContent
+          children={children as any}
+          className={className}
+          ref={ref}
+        />
+      </CoreForm.Root>
+    );
+  },
+);
+
+/**
+ * Internal component to handle the Root content with service access
+ */
+const RootContent = React.forwardRef<
+  HTMLDivElement,
+  { children: React.ReactNode; className?: string }
```

Comment:
Since you render a div, add asChild support, and the prop types should be any prop allowed on a div so you should get the ref, children and className and use them all
`const { children, className, ...otherProps } ...` and `<div ref={ref} className={className} {...otherProps}>`